### ISSUE TITLE
[3.2] "Mein Leben" skill

### ISSUE THREAD

**Issue by MuadDib90** [Realm667/WolfenDoom#952]
Side quest, there is an old radio from WWII. If it is on purpose, like they studing it, then OK. Otherwise it is out of place.
![01](https://user-images.githubusercontent.com/73436978/125654339-ac9d6ab4-0197-4386-a74b-6b4cd9633495.PNG)

Then there is the question of Mein Leben skill.
It's like maybe 2 to 3 months I played it on this skill, I wanted to test also the others a bit.
But there had to be some heavy modifications on this.

First, the new unit Ursula Mertzger (or something like that), seems abnormally hard to beat; compared to other bosses. I spent about 40 minutes on her. Constantly loading, because her attacs were very fast, and I was counting every hit on her. And if someone do not save ammo as I did, this is easily a dead end.

Secondly, 1 hour, one whole hour and I was unable to get through Tiger Roboter arena. To be precise, I was unable even to get them any harm, and save my progress, to be able to load and try again as in the case with Ursula. They have now (aside from their primary attack of bumbing green balls) also some sort of rapid "gun fire" (those small yellow? stars that run toward you very fast), which is literally unable to outmaneuver (considering there are 4 of them at start).

With all of this I am only imagining the final battle agains Cyber Hitler (because I couldn't pass through to him) being nothing less than impossible.

Final thought, I am not sure, if this is only on Mein Leben, but now it gets really frustrating. Don't worry, I know, some things must be done to be it some sort of challenge, because you play agains "stupid NPCs", not agains actual people, but when you try to lean around corner to hit something and immediately hide behind it, to be able to shoot down someting, it gets frustrating, because NPCs tend to start shooting and also hit you immediately after you show your first hitbox pixel from behind corner! You have regulary zero chance to do something. Nobody has that kind of reflexies. You try hard to kill something, but every time you "show yourself", you get hit no matter what. And you can't even "bring in" the tactic to get hurt to be able to shoot "it", because you have so much less HP than that NPC you're trying to kill, so you stand no chance. So only luck and many many loads. Nobody's can possibly have fun with this.

---

In conclusion:
Please, take my advice. I dare to say, nobody played and tried to test it on Mein Leben longer than me. And I completed that more than once. So if "I" can't do it, there has to be something wrong. (Or is there something I am missing...? EDIT: Or anyone else probably did it? If so, I sure as hell would like to see that. :)

**Comment by Tormentor667** [comment]
I totally agree with you regarding this. I think we need to change the difficulty curve in general for „Mein Leben“ and the higher skill Settings, not only for Astrostein or specific enemies.

**Comment by MuadDib90** [comment]
> I totally agree with you regarding this. I think we need to change the difficulty curve in general for „Mein Leben“ and the higher skill Settings, not only for Astrostein or specific enemies.

How do we do that?

**Comment by Tormentor667** [comment]
The skill settings are defined in the MAPINFO lump from line 579 onwards. If I understand it correctly, skill settings 1-3 are well balanced, skill settings 4-5 are too hard, is that right?

**Comment by MuadDib90** [comment]
> The skill settings are defined in the MAPINFO lump from line 579 onwards. If I understand it correctly, skill settings 1-3 are well balanced, skill settings 4-5 are too hard, is that right?

Tbh I've never played on 4th skill (maybe with some minor occasions). Most of the time I played on 5. So can't really speak for 4, but 5 was half a year back really "not good". Pretty hard in general and especially Astrostein was unbeatable for me.

-- 

So, how do you define the defficulty? With variables for enemies (strength, armor, speed, their number ect.) set to "some high" given the user input of skill setting?

**Comment by Tormentor667** [comment]
Afaik, the skill settings are defined by multipliers and different projectile actors for different skill settings.

I think reducing damage in the higher settings would already help a lot, as well as reducing projectile speeds.

**Comment by MuadDib90** [comment]
Also regarding this... As I stated in the first paragraph, I am being constantly irritated (a lot) by enemies reaction time on Mein Leben skill. I am voting to tone down this in the first place. 

Imagine there is an enemy behind the corner and you know about him. You try to sneak around corner and try to hit him the first; but you can't. You just can't, because their reaction time is impossibly quick. They hit you always the first. It's frustrating, no matter how hard you try.

Thus you are unable to go slowly (if you want to that is), trying to surprise enemies around corners, because you are constantly hit by bullets. (Not talking about "raid", going straight to an enemy in doom style shooting everything around you in run.)

**Comment by Username-N00b-is-not-available** [comment]
> Imagine there is an enemy behind the corner and you know about him. You try to sneak around corner and try to hit him the first; but you can't. You just can't, because their reaction time is impossibly quick. They hit you always the first.

Is this related to Astrostein enemies, or to every enemy? It is, indeed, possible on Mein Leben to shoot first and lock in a pain state even an already aiming guard. You just have to aim at his location when still behind cover (so this skill is not for first-time players anyway). The skill is supposed to be very hard, and increasing enemy reaction time will just make it a copy of the penultimate skill. But it **is** possible to avoid damage in most situations. You can save high-damage ammo types for dangerous enemies and use obstacles and Doom engine oddities to reduce chances of bullets hitting you.
On open maps you are of course out of luck. I also remember the Roboter fight being exceptionally hard on Mein Leben, but that was in 2021, I do not remember details.

**Comment by Username-N00b-is-not-available** [comment]
The C2M0_B map on Mein Leben is currently pure frustration. I have not finished it yet, here are some initial observations.
1. The quad-pistol robots are too OP for a melee player. If you decide to engage them with the Elektroschocker (even behind a corner), expect to lose at least 40 damage or just to reload the savegame once again. To make matters worse, some of the enemies are static and won't move unless you have damaged them --- i. e. either used a grenade or came up close --- but in the latter case you are dead in a matter of seconds.
Here, a bit later, you can at least hide in the crater and quickly uncrouch/shoot/crouch them, which is a nice bit of tactics:
![image](https://user-images.githubusercontent.com/59409713/233830725-45c16366-9afe-448b-b278-ff692caca9e3.png)
Hiding in the empty corridors is not possible at all (only behind corners), not counting that you don't even have a ranged weapon.
2. The doors on SS Eisenherz are closing too fast, you have to spam the switch every three to five seconds. Though is is a minor nuisance compared to 1.
3. IIRC in the city there are some repopulation scripts for fodder pistol enemies. For a Mein Leben player this might turn from an artistic touch into a serious problem, since there is not nearly enough ammo to kill them without health loss. There is generally too little ammo for the pistol, and not that much shotgun ammo that it could be wasted on large distance targets.

My suggestions for now:
- add a double ammo factor to Mein Leben, including grenades (for all levels); the fights will resemble a real gunfight, where ammo is abundant and not hitting an enemy is no big deal compared to being hit yourself;
- considerably increase pain chance for Elektroschocker, maybe even add +FORCEPAIN on the puff.

By the way, Astrostein ammo is not shown on the Doom-style statusbar, it should be possible to draw it there (or even replace it with the original Laz Rojas statusbar). I can volunteer to change the code, if necessary.
What do you think @Ozymandias81 @Tormentor667?

**Comment by Username-N00b-is-not-available** [comment]
The Roboter and Mecha-Hitler battles are
### _`   I   N   S   A   N   I   T   Y   .   `_
These kind of battles make the player ragequit and delete the game, the difficulty spikes will definitely not be manageable even for the most experienced players.

The Roboter fight is exactly the same BS as it was 2 years ago. I don't see how these homing missiles are to be avoided by a human player. They turn very fast, and on Mein Leben also obtain high velocity. The only way to eliminate them is to crash them into the raising pillar with perfect timing, but this is way easier said than done, especially since you cannot regenerate stamina that fast.
The later encounters with Roboters are fine, near and in the K. I. Kortex you can just shotgun the Roboters from behind cover.

The Mecha-Hitler fight is just a spam of bullet sponges that spam the screen with small projectiles of Doom II Revenant damage (thanks that they are not homing as with the Roboters, at least!) --- and Hitler is also a really big bullet sponge. Notwithstanding that it can still be beaten, it is even worse than the Roboter fight. The real battle here goes between the player and ```frandom()``` in projectile trajectories and damage (the player WILL have to take it). And killing the ```AstroCyborg2```'s that are in your path is not really an option, since you will spend too much time and new ones will spawn.

Here are some [savegames](https://github.com/Realm667/WolfenDoom/files/11304645/C2M0B_MLsave.zip) in case anyone is interested. I retract my statement on repopulation-script spawned enemies; compared to the boss fights, these do not matter at all.

So my further suggestions are
- lower enemy health by 50% on Mein Leben;
- slightly increase the quad-pistol ```AstroSuiteGuard``` reaction time;
- do something to the final battle? No idea on how to improve it at the moment without making it too easy on lower skills.

**Comment by Tormentor667** [comment]
Good point that someone is coming up with this.

I agree with everything and my suggestions are

- Reduce enemy health on Mein Leben
- Reduce robot spawnings on the final mecha-hitler battle
- Make robot projectiles home only for some time and then stop homing, so you have to avoid them for a few seconds but then they let go of you

**Comment by Username-N00b-is-not-available** [comment]
@Tormentor667 Thanks! Though the Roboter projectiles won't live that long anyway, the problem is that even a sprinting player has tough luck avoiding them when traveling from one pillar to another. I think it would be better just to align their speed on Mein Leben with speeds on other skill levels.
What do you think about the following suggestions from my first message (you only commented on the second):
> - add a double ammo factor to Mein Leben, including grenades (for all levels); the fights will resemble a real gunfight, where ammo is abundant and not hitting an enemy is no big deal compared to being hit yourself;
> - considerably increase pain chance for Elektroschocker, maybe even add +FORCEPAIN on the puff.

**Comment by Tormentor667** [comment]
I am okay with these changes, good points!

**Comment by MuadDib90** [comment]
I am really glad, that you know now I didn't just blather... but my points were serious, and - of course - you are able to do something with it. 👍
